core/src/avm2/domain.rs

```diff
@@ -383,8 +427,14 @@ impl<'gc> Domain<'gc> {
 
         Ok(())
     }
+
+    pub fn as_ptr(self) -> *const DomainPtr {
+        self.0.as_ptr() as _
+    }
 }
 
+pub enum DomainPtr {}
```

Comment:
I'm not seeing the entire codebase right now, but is this pattern used in other places as well? If so, that's fine, I'm just not familiar with it. Otherwise, a comment would be nice about what this is, and why is it the way it is.

Reply:
It's used in quite some places